_posts/2021-04-07-upgrade-improvements.md

```diff
@@ -0,0 +1,63 @@
+The main improvement is, that you don't need to upgrade across all Strimzi versions, but you can upgrade from for instance 0.18 directly to 0.22 as it is shown in the diagram bellow.
+You can face the following states:
+* Supported Kafka `version` and `log.message.format.version` or `inter.broker.protocol.version` is set in CR - Strimzi will keep the values and missing one will se based on `version`.
+* Supported Kafka `version` is set, `log.message.format.version` and `inter.broker.protocol.version` are not set in CR - Strimzi will set `log.message.format.version` and `inter.broker.protocol.version` based on `version`
+* Supported Kafka `version` is not set in CR - Strimzi will set Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` based on the default Kafka supported in 0.22.
+* Unsupported Kafka `version` is set in CR - Strimzi will set `Kafka` CR status to `NotReady`. 
+  In that case user has to update CR and manually set a proper config.
+
+Strimzi now does the following things regarding Kafka configuration:
+* Detect used Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` (from CR or compute it from Kafka `version`).
+* Configure Kafka `version`, `log.message.format.version` and `inter.broker.protocol.version` if it's not specified by user.
+
+Users don't need to fill Kafka `version`, but also `log.message.format.version` or `inter.broker.protocol.version` before the upgrade, but we strongly recommend it.
+
+#### Upgrade without versions
+
+Strimzi is able to deal with upgrade of Kafka clusters where versions are not set in custom resource.
+In cases like that, Strimzi is able to determine the versions and go through the upgrade as follows:
+* Get version of currently deployed Kafka and set it in custom resource.
+* Perform Kafka cluster rolling update to change the images from old Strimzi version to new one.
+* Upgrade Kafka `version` to default version support by new Strimzi version which will perform rolling update.
+* Get `log.message.format.version` and `inter.broker.protocol.version` based on `version`.
+* Set `log.message.format.version` and `inter.broker.protocol.version` and perform Kafka cluster rolling update.
+
+You can see the states of the upgrade process and each rolling update in the diagram bellow.
+
+![Kafka upgrade states](/assets/images/posts/2021-04-19-kafka-rolling-updates.png)
+
+#### Disadvantages
+
+Even with these changes, there are still some disadvantages:
+* For upgrade, roll out the new brokers while first using the older `log.message.format.version` or `inter.broker.protocol.version` and only afterwards change to the new versions for message format and inter-broker protocol.
+* Downgrade won't be executed when new `log.message.format.version` or `inter.broker.protocol.version` are already used.
+  You need to change configuration to use same or older1 `log.message.format.version` and `inter.broker.protocol.version` as older Kafka.
```

Comment:
I think this is not so simple as in this case the downgrade might not be possible anymore (cc @tombentley).

Reply:
Discussed offline and I did some changes in this part according to the docs.